DirectProgramming/DPC++FPGA/ReferenceDesigns/mvdr_beamforming/src/NullPipe.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef __NULL_PIPE_HPP__
+#define __NULL_PIPE_HPP__
+
+// NullPipe
+// Intended to connect to kernels that have unneeded pipes.
+// Blocking writes accept and discard data and return immediately.
+// Non-blocking writes accept and discard data and return immediately with
+// success = true.
+// Blocking reads never return.
+// Non-blocking reads return immediately with success = false.
+template <
+  class     Id,   // name of the pipe
+  typename  T     // data type accepted by the pipe
+> struct NullPipe {
+
+  NullPipe() = delete;  // ensure we cannot create an instance
+
+  // Non-blocking write
+  static void write( const T & /*data*/, bool &success ) {
+    success = true;
+  }
+
+  // Blocking write
+  static void write( const T & /*data*/ ) {
+    // do nothing
+  }
+
+  // Non-blocking read
+  static T read( bool &success ) {
+    T return_value = T();   // call the default constructor to avoid warnings
+                            // about uninitialized variables
+    success = false;
+    return( return_value );
+  }
+
+  // Blocking read
+  static T read() {
+    T return_value = T();   // call the default constructor to avoid warnings
+                            // about uninitialized variables
+    while(1) {
```

Comment:
Why is an infinite loop correct here?

Reply:
All of the 'processing' kernels have infinite loops.  This design is a streaming design, meaning it is processing a constant stream of data from some external source (normally a network port or some other external IO, in our case we have to 'fake' this with pipes that are fed from the host since we don't have a board in the dev cloud with real IO pipes).  As such, there is no clear point when the kernel should stop.  The idea is the host program launches them and then they just process any data they receive indefinitely.